eng/pipelines/templates/steps/npm-release-task.yml

```diff
@@ -4,61 +4,120 @@ parameters:
+              echo "##vso[task.setvariable variable=IntendedTagVersion;isOutput=true]$packageVersion"
+            name: SetPackageVar
+            displayName: Get original tags
+
+          - task: EsrpRelease@9
+            displayName: 'Publish ${{parameters.Artifact.name}} to ESRP'
+            inputs:
+              ConnectedServiceName: 'Azure SDK PME Managed Identity'
+              ClientId: '5f81938c-2544-4f1f-9251-dd9de5b8a81b'
+              DomainTenantId: '975f013f-7f24-47e8-a7d3-abc4752bf346'
+              Usemanagedidentity: true
+              KeyVaultName: 'kv-azuresdk-codesign'
+              SignCertName: 'azure-sdk-esrp-release-certificate'
+              Intent: 'PackageDistribution'
+              ContentType: 'npm'
+              FolderLocation: ${{parameters.PathToArtifacts}}
+              Owners: ${{ coalesce(variables['Build.RequestedForEmail'], 'azuresdk@microsoft.com') }}
+              Approvers: ${{ coalesce(variables['Build.RequestedForEmail'], 'azuresdk@microsoft.com') }}
+              ServiceEndpointUrl: 'https://api.esrp.microsoft.com'
+              MainPublisher: 'ESRPRELPACMANTEST'
+              productstate: ${{parameters.Tag}}
 
+        - ${{ else }}:
+          - task: PowerShell@2
+            displayName: Publish to Dev Feed
+            inputs:
+              targetType: filePath
+              filePath: eng/tools/publish-to-npm.ps1
+              arguments: -pathToArtifacts ${{parameters.PathToArtifacts}} -accessLevel "public" -tag "${{parameters.Tag}}" -additionalTag "${{parameters.AdditionalTag}}" -registry ${{parameters.Registry}} -npmToken $(azure-sdk-npm-token)
+              pwsh: true
+            condition: succeeded()
+
+- ${{ if eq(parameters.Registry, 'https://registry.npmjs.org/') }}:
+  - job: VerifyPackageTags
+    dependsOn: PublishPackage
+    condition: succeeded()
+    variables:
+      PackageName: $[dependencies.PublishPackage.outputs['SetPackageVar.PackageName']]
+      OriginalTags: $[dependencies.PublishPackage.outputs['SetPackageVar.OriginalTags']]
+      IntendedTagVersion: $[dependencies.PublishPackage.outputs['SetPackageVar.IntendedTagVersion']]
+    displayName: 'Verify package tags'
+    pool:
+      name: azsdk-pool-mms-ubuntu-2004-general
+      image: azsdk-pool-mms-ubuntu-2004-1espt
+      os: linux
+    steps:
+    - checkout: self
     - task: Powershell@2
       displayName: Verify package tags
       inputs:
         targetType: filePath
         filePath: eng/scripts/verify-npm-tags.ps1
-        arguments: -originalDistTags '$(OriginalTags)' -intendedTag '${{parameters.Tag}}' -intendedTagVersion '$(IntendedTagVersion)' -packageName $(PackageName) -npmToken '$(azure-sdk-npm-token)'
+        arguments: -originalDistTags '$(OriginalTags)' -intendedTag '${{parameters.Tag}}' -intendedTagVersion '$(IntendedTagVersion)' -packageName '$(PackageName)' -npmToken '$(azure-sdk-npm-token)'
```

Comment:
Is this supposed to handle a case where `PackageName` or other variables aren't set? We should probably use double quotes so that powershell will fail out in that case (since they will be the literal $() syntax which is invalid).